packages/slate-editor/src/components/Portals/BasePortalV2.tsx

```diff
@@ -75,7 +75,22 @@ const BasePortalV2: FunctionComponent<Props> = ({
         if (rect === null) {
             setReferenceElement(null);
         } else {
-            const clientRect = { x: 0, y: 0, toJSON: () => null, ...rect };
+            /**
+             * We have to manually re-create a ClientRect-shape object instead of `...rect`,
+             * as `DOMRect` object properties are not enumerable.
+             * @see https://github.com/microsoft/TypeScript/issues/9726
```

Comment:
:open_mouth: 

Good job nailing it down :+1: 

Since this is a open source repo, maybe we can share this as an example of where it is not an [_edge case_](https://github.com/microsoft/TypeScript/issues/9726#issuecomment-240785269)?